doc/adr/0013-download-for-incremental-snapshots.md

```diff
@@ -0,0 +1,91 @@
+
+The modified download process involves several steps to ensure that nodes have a complete and validated chain of incremental snapshots. The first step is to determine the starting point by querying a peer node for the latest snapshot hash. 
+
+```mermaid
+flowchart TD
+  subgraph Determine starting point
+    A[Get latest snapshot from the Joinee]
+  end
+  subgraph Download process
+	B[Get snapshot with hash X=aaaa]
+  end
+  A --> |latest hash: aaaa| B
+```
+
+The node then checks if it has all the incremental snapshots needed to reach the starting point, and if not, it beings downloading missing snapshots recursively from the latest snapshot to the genesis. All validated snapshots are stored in permanent storage, which contains a correct and validated chain of incremental snapshots. Temporary storage may contain potentially invalid or conflicting snapshots.
+
+Once all the incremental snapshots have been downloaded, the node performs chain validation by applying subsequent incremental snapshots from the genesis to the latest snapshot to build the network state and validate the chain. If the latest snapshot on the network has been updated and the batch is not small enough (determined by the `k` parameter), the node repeats the download process for the next batch of snapshots. Otherwise, it can register for consensus.
+
+```mermaid
+flowchart TD
+  subgraph Download process
+	  A([Get snapshot with hash X])
+	  A --> B
+	  B{"check if X is on disk<br/>in perm-storage<br/>(validated)"}
+	  B --> |"Yes<br/><br/>pass map Ordinal->Hash for data from temp-storage"| C
+	  B --> |No| D
+
+	  subgraph validation [Chain validation]
+	    C[[start chain validation]]
+	    C1([end process - critical error])
+
+	    C2{"is batch of size 1 or less<br/>(k parameter)"}
+	    C --> |fails| C1
+	    C --> |succeed| C2
+
+	    C2C1([register for consensus])
+	    C2C2{{repeat process for next batch}}
+	    C2 --> |Yes| C2C1
+	    C2 --> |No| C2C2
+	  end
+	  
+	  C2C2 -.-> A
+
+	  subgraph download [Download]
+	    D{check if X<br/>is in temp storage}
+	    E[read from temp-storage]
+	    F[download X from network]
+
+	    D --> |Yes| E
+	    D --> |No| F
+
+	    G{validate that hash bytes matches X}
+	    F --> G
+
+	    H[save it to temp-storage]
+
+	    I{{repeat process for X.lastSnapshotHash,<br/>unless X is full snapshot}}
+
+	    E --> I
+	    H --> I
+
+	    J[retry download from different source<br/>if validation failed]
+	    G --> |Yes| H
+	    G --> |No| J
+	    J --> F
+	  end
+	  
+	  I -.-> A
+  end
+```
+
+This modified download process ensures that all joining nodes have a complete and validated chain of incremental snapshots and are able to determine the latest state of the network accurately. Although the initial node joining process may take longer, it ensures consistency and accuracy across all nodes and mitigates the risk of inconsistent and invalid data.
```

Comment:
What happens if there is a new snapshot that has been added while the new node is recursively downloading the incremental snapshots? How is that captured?

Reply:
It's captured and described below. Node performs download for another batch as long as the batch is small enough to register for consensus. 

> If the latest snapshot on the network has been updated and the batch is not small enough (determined by the k parameter), the node repeats the download process for the next batch of snapshots. Otherwise, it can register for consensus.